### PR TITLE
arch: arm: soc: nordic: Migrate to new logging subsys

### PR DESCRIPTION
**Pull request by Olivier-ProGlove**
Migrate from SYS_LOG to LOG logging mechanism.

Signed-off-by: Olivier Martin <olivier.martin@proglove.de>

### CODE REVIEW

File: arch/arm/soc/nordic_nrf/nrf52/power.c [hunk 1]
```diff
@@ -49,7 +51,7 @@ static void _low_power_mode(enum power_states state)
 
 	default:
 		/* Unsupported State */
-		SYS_LOG_ERR("Unsupported State\n");
+		LOG_ERR("Unsupported State\n");
```
Comment (nordic-krch) [inline arch/arm/soc/nordic_nrf/nrf52/power.c]: newline not needed. logger adds it.

File: arch/arm/soc/nordic_nrf/nrf52/power.c [hunk 2]
```diff
@@ -76,7 +78,7 @@ void _sys_soc_set_power_state(enum power_states state)
 #endif
 	default:
 		/* Unsupported State */
-		SYS_LOG_ERR("Unsupported State\n");
+		LOG_ERR("Unsupported State\n");
```
Comment (nordic-krch) [inline arch/arm/soc/nordic_nrf/nrf52/power.c]: ditto

File: arch/arm/soc/nordic_nrf/nrf52/power.c [hunk 3]
```diff
@@ -96,7 +98,7 @@ void _sys_soc_power_state_post_ops(enum power_states state)
...
```
Comment (nordic-krch) [inline arch/arm/soc/nordic_nrf/nrf52/power.c]: ditto

File: arch/arm/soc/nordic_nrf/nrf52/power.c [hunk 4]
```diff
@@ -4,8 +4,10 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define LOG_MODULE_NAME soc_power
```
Comment (pizi-nordic) [inline arch/arm/soc/nordic_nrf/nrf52/power.c]: What about nrf_power? Such name will suggest where look for the code which sent the message.

Comment (ioannisg) [reply arch/arm/soc/nordic_nrf/nrf52/power.c]: or nrf_soc_power

**Comment by ioannisg** [comment]
@Olivier-ProGlove just wondering: power.c is the only Nordic SOC file that contains log messages?

**Review by ioannisg — COMMENTED** [review]
LGTM. I am wondering whether other Nordic SOC source files have SYS log messages that should migrate to the new logging sub system.

**Comment by Olivier-ProGlove** [comment]
@pizi-nordic @nordic-krch @ioannisg Comments addressed!

**Comment by Olivier-ProGlove** [comment]
@ioannisg I forgot to answer your question, here is the grep on my pull-request rebased on the current head of `master`:

```
$ grep -r LOG arch/arm/soc/
arch/arm/soc/nxp_imx/mcimx7_m4/soc_clk_freq.c:		hz = CCM_ANALOG_GetSysPllFreq(CCM_ANALOG) >> 2;
arch/arm/soc/nxp_kinetis/kwx/wdog.S:_ASM_FILE_PROLOGUE
arch/arm/soc/nxp_kinetis/k6x/wdog.S:_ASM_FILE_PROLOGUE
arch/arm/soc/nordic_nrf/nrf52/power.c:#define LOG_MODULE_NAME nrf_soc_power
arch/arm/soc/nordic_nrf/nrf52/power.c:#define LOG_LEVEL LOG_LEVEL_DEBUG
arch/arm/soc/nordic_nrf/nrf52/power.c:LOG_MODULE_REGISTER();
arch/arm/soc/nordic_nrf/nrf52/power.c:		LOG_ERR("Unsupported State");
arch/arm/soc/nordic_nrf/nrf52/power.c:		LOG_ERR("Unsupported State");
arch/arm/soc/nordic_nrf/nrf52/power.c:		LOG_ERR("Unsupported State");
arch/arm/soc/ti_lm3s6965/soc.h:#define IRQ_ANALOG_COMP0 25
arch/arm/soc/ti_lm3s6965/soc.h:#define IRQ_ANALOG_COMP1 26
```

Actually looking at the result, the line `#define LOG_LEVEL LOG_LEVEL_DEBUG` is not needed.  I will remove it in the following pull-request.

**Review by ioannisg — APPROVED** [review]
Approving this. @Olivier-ProGlove have you tested it, btw?

**Comment by nashif** [comment]
addressed in #10029

**Pull request closed without merging**